cmd/kubelet/app/plugins.go

```diff
@@ -44,3 +48,14 @@ func ProbeVolumePlugins() []volume.Plugin {
 
 	return allPlugins
 }
+
+// ProbeNetworkPlugins collects all compiled-in plugins
+func ProbeNetworkPlugins() []network.NetworkPlugin {
+	allPlugins := []network.NetworkPlugin{}
+
+	// for each existing plugin, add to the list
+	allPlugins = append(allPlugins, exec.ProbeNetworkPlugins()...)
+	allPlugins = append(allPlugins, no_op.ProbeNetworkPlugins()...)
```

Comment:
I don't like that the no-op mechanism is buried deep down, but depends on this layer including no-op.  I'd prefer that the no-op plugin be "special".  Don't probe for it, just define it inline with the plugin init code and if the --network-plugin flag is "", instantiate the no-op.  It's special.